caffe2/core/context_gpu.cu

```diff
@@ -428,6 +428,24 @@ CUDAContext::CUDAContext(const DeviceOption& option)
   DCHECK_EQ(option.device_type(), PROTO_CUDA);
 }
 
+CUDAContext::~CUDAContext() {
+  try {
+    if (curand_generator_) {
+      CURAND_CHECK(curandDestroyGenerator(curand_generator_));
+    }
+    // CUDAContext is used in 2 cases now:
+    // - long-lived instance inside OperatorBase in which case what happens in
+    //   destructor doesn't really matter
+    // - short-lived on-the-fly instances that are utilized as CUDAGuard - in
+    //   this case there's only one stream id (passed to SwitchToDevice) and
+    //   it's preferrable to synchronize in the destructor
+    FinishDeviceComputation();
+  } catch (const std::exception& e)  {
+    LOG(FATAL) << "Encountered following in " << __FUNCTION__ << ": " << e.what();
```

Comment:
This doesn't seem like an improvement, doesn't FATAL also terminate the program too?

Reply:
Sorry, forgot to update the PR with one that reduces the severity to "LOG(ERROR)". 
I wonder if better fix would be to declare it with `noexcept(false)` modifier

Reply:
Tried that, but it looks too ugly (need to declare both OperatorBase and Observable with `noexcept(false)`) so I'm going to land this PR (but reduce the log level to ERROR